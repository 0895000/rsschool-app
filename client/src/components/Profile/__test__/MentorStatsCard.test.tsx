import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import MentorStatsCard from '../MentorStatsCard';

describe('MentorStatsCard', () => {
  const mentorStats = [
    {
      courseName: 'rs-2020-q1',
      locationName: 'Minsk',
      courseFullName: 'Rolling Scopes School 2020-Q1',
    },
    {
      courseName: 'rs-2018-q1',
      locationName: 'Minsk',
      courseFullName: 'Rolling Scopes School 2018-Q1',
      students: [
        {
          githubId: 'alex',
          name: 'Alex Petrov',
          isExpelled: false,
          totalScore: 3453,
        },
        {
          githubId: 'vasya',
          name: 'Vasiliy Alexandrov',
          isExpelled: true,
          totalScore: 120,
        },
      ],
    },
  ];
  describe('Should render correctly', () => {
    it('if is editing mode disabled', () => {
      const output = shallow(
        <MentorStatsCard data={mentorStats} isEditingModeEnabled={false} onPermissionsSettingsChange={jest.fn()} />,
      );
      expect(shallowToJson(output)).toMatchSnapshot();
    });
    it('if is editing mode enabled', () => {
      const output = shallow(
        <MentorStatsCard data={mentorStats} isEditingModeEnabled={true} onPermissionsSettingsChange={jest.fn()} />,
      );
      expect(shallowToJson(output)).toMatchSnapshot();
    });
  });

  const wrapper = shallow(
    <MentorStatsCard data={mentorStats} isEditingModeEnabled={false} onPermissionsSettingsChange={jest.fn()} />,
  );
  const instance: any = wrapper.instance();
  describe('showMentorStatsModal', () => {
    it('should set "state.isMentorStatsModalVisible" as "true", "state.courseIndex" as index was passed', () => {
      expect(instance.state.isMentorStatsModalVisible).toBe(false);
      expect(instance.state.courseIndex).toBe(0);
      instance.showMentorStatsModal(1);
      expect(instance.state.isMentorStatsModalVisible).toBe(true);
      expect(instance.state.courseIndex).toBe(1);
    });
  });
  describe('hideMentortStatsModal', () => {
    it('should set "state.isMentorStatsModalVisible" as "false"', () => {
      instance.state.isMentorStatsModalVisible = true;
      expect(instance.state.isMentorStatsModalVisible).toBe(true);
      instance.hideMentortStatsModal();
      expect(instance.state.isMentorStatsModalVisible).toBe(false);
    });
  });
  describe('filterPermissions', () => {
    it('should left only "isProfileVisible" in "permissionsSettings" object', () => {
      const permissionsSettings = {
        isProfileVisible: { all: true },
        isAboutVisible: { all: true, mentor: true, student: true },
        isEducationVisible: { all: true, mentor: true, student: true },
        isEnglishVisible: { all: false, student: false },
        isEmailVisible: { all: true, student: true },
        isTelegramVisible: { all: false, student: false },
        isSkypeVisible: { all: true, student: true },
        isPhoneVisible: { all: false, student: false },
        isContactsNotesVisible: { all: true, student: true },
        isLinkedInVisible: { all: false, mentor: false, student: false },
        isPublicFeedbackVisible: { all: true, mentor: true, student: true },
        isMentorStatsVisible: { all: true, mentor: true, student: true },
        isStudentStatsVisible: { all: true, student: true },
      };
      const instance: any = wrapper.instance();
      const result = instance.filterPermissions(permissionsSettings);
      expect(result).toEqual({
        isMentorStatsVisible: { all: true, mentor: true, student: true },
      });
    });
  });
});
