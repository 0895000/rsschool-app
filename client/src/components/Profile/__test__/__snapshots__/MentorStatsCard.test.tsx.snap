// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MentorStatsCard Should render correctly if is editing mode disabled 1`] = `
<Fragment>
  <MentorStatsModal
    isVisible={false}
    onHide={[Function]}
    stats={
      Object {
        "courseFullName": "Rolling Scopes School 2020-Q1",
        "courseName": "rs-2020-q1",
        "locationName": "Minsk",
      }
    }
  />
  <CommonCard
    content={
      <React.Fragment>
        <div>
          <p>
            Mentored Students:
             
            <Text
              strong={true}
              style={
                Object {
                  "fontSize": 18,
                }
              }
            >
              3
            </Text>
          </p>
          <p>
            Courses as Mentor:
             
            <Text
              strong={true}
              style={
                Object {
                  "fontSize": 18,
                }
              }
            >
              2
            </Text>
          </p>
        </div>
        <List
          bordered={false}
          dataSource={
            Array [
              Object {
                "courseFullName": "Rolling Scopes School 2020-Q1",
                "courseName": "rs-2020-q1",
                "locationName": "Minsk",
              },
              Object {
                "courseFullName": "Rolling Scopes School 2018-Q1",
                "courseName": "rs-2018-q1",
                "locationName": "Minsk",
                "students": Array [
                  Object {
                    "githubId": "alex",
                    "isExpelled": false,
                    "name": "Alex Petrov",
                    "totalScore": 3453,
                  },
                  Object {
                    "githubId": "vasya",
                    "isExpelled": true,
                    "name": "Vasiliy Alexandrov",
                    "totalScore": 120,
                  },
                ],
              },
            ]
          }
          itemLayout="horizontal"
          loading={false}
          pagination={false}
          renderItem={[Function]}
          split={true}
        />
      </React.Fragment>
    }
    icon={<ForwardRef(TeamOutlined) />}
    isEditingModeEnabled={false}
    onPermissionsSettingsChange={[MockFunction]}
    title="Mentor Statistics"
  />
</Fragment>
`;

exports[`MentorStatsCard Should render correctly if is editing mode enabled 1`] = `
<Fragment>
  <MentorStatsModal
    isVisible={false}
    onHide={[Function]}
    stats={
      Object {
        "courseFullName": "Rolling Scopes School 2020-Q1",
        "courseName": "rs-2020-q1",
        "locationName": "Minsk",
      }
    }
  />
  <CommonCard
    content={
      <React.Fragment>
        <div>
          <p>
            Mentored Students:
             
            <Text
              strong={true}
              style={
                Object {
                  "fontSize": 18,
                }
              }
            >
              3
            </Text>
          </p>
          <p>
            Courses as Mentor:
             
            <Text
              strong={true}
              style={
                Object {
                  "fontSize": 18,
                }
              }
            >
              2
            </Text>
          </p>
        </div>
        <List
          bordered={false}
          dataSource={
            Array [
              Object {
                "courseFullName": "Rolling Scopes School 2020-Q1",
                "courseName": "rs-2020-q1",
                "locationName": "Minsk",
              },
              Object {
                "courseFullName": "Rolling Scopes School 2018-Q1",
                "courseName": "rs-2018-q1",
                "locationName": "Minsk",
                "students": Array [
                  Object {
                    "githubId": "alex",
                    "isExpelled": false,
                    "name": "Alex Petrov",
                    "totalScore": 3453,
                  },
                  Object {
                    "githubId": "vasya",
                    "isExpelled": true,
                    "name": "Vasiliy Alexandrov",
                    "totalScore": 120,
                  },
                ],
              },
            ]
          }
          itemLayout="horizontal"
          loading={false}
          pagination={false}
          renderItem={[Function]}
          split={true}
        />
      </React.Fragment>
    }
    icon={<ForwardRef(TeamOutlined) />}
    isEditingModeEnabled={true}
    onPermissionsSettingsChange={[MockFunction]}
    title="Mentor Statistics"
  />
</Fragment>
`;
